### PR TITLE
[Google Blockly] Turn on scrollbars

### PR DESCRIPTION
**Pull request by ajpal**
Adds blockly option to enable workspace scrollbars. 
Customizations needed:
- Hide the horizontal scrollbar. Doing this mostly to match our current Blockly behavior.
- Hide the vertical scrollbar when it's not needed
Oct-20-2020 15-48-09

- Prevent workspace from scrolling in all directions:
Blockly treats the workspace as infinitely expandable in every direction, with the origin in the middle. As a result, the default view starts halfway down the page:
image
so that you can scroll up and to the left of your first block:
image

We want the first block to always sit in the top-left corner of the workspace. To do this, I just had to override the `getContentDimensionsBounded_` function. This is a private function, but there's no officially-supported way to get the functionality we want. This isn't our first customization of a private function, so we'll have to be careful when updating versions regardless.
image

- The other customization I had to make was for blocks being placed in negative locations. With the infinite-scroll workspace, negative block coordinates are perfectly valid, but since we keep (0,0) locked to the top-left corner, we don't want to allow placing blocks in negative locations. This was causing weird behavior when you tried to delete undeletable blocks- instead of being bumped back into the workspace, they were being placed underneath the toolbox (since the negative X location was treated as valid). To fix this, I added a check to the beginning of `endBlockDrag` to bump the block back into view.
Before:
Oct-19-2020 16-00-21

After:
Oct-19-2020 15-58-34

This PR also fixes an issue where the bottom of the workspace was getting clipped:
Before:
image

After:
image

### CODE REVIEW

File: apps/src/blocklyAddons/cdoWorkspaceSvg.js [hunk 1]
```diff
@@ -70,3 +70,22 @@ WorkspaceSvg.prototype.blockSpaceEditor = {
...
+  const content = Blockly.WorkspaceSvg.getContentDimensionsExact_(ws);
+
+  // View height and width are both in pixels, and are the same as the SVG size.
+  const viewWidth = svgSize.width;
+  const viewHeight = svgSize.height;
+
+  const dimensions = {
+    left: 0,
+    top: 0,
+    height: Math.max(content.bottom + viewHeight / 2, viewHeight),
```
Comment (breville) [inline apps/src/blocklyAddons/cdoWorkspaceSvg.js]: Out of curiosity, what is this doing?

Comment (ajpal) [reply apps/src/blocklyAddons/cdoWorkspaceSvg.js]: the first term is the length of the blocks in the workspace + a little bit (this is how we make more room as your code gets close to the bottom). The second term is just the height of the visible space. The workspace height should be the max of these two. In other words, the workspace fits the height of its container, unless the code is long, in which case it is the height of the code plus a buffer.

Comment (breville) [reply apps/src/blocklyAddons/cdoWorkspaceSvg.js]: Interesting!  Given there is a big of artistic license there, maybe a comment to capture the thinking?

**Review by breville — APPROVED** [review]

**Review by breville — APPROVED** [review]
Thanks for the detailed comments!

**Pull request merged**